File: benchmarks/tsb/bench_series_describe.ts
/**
 * Benchmark: describe(s) — summary statistics function on 100k Series.
 */
import { Series, describe } from "../../src/index.js";

const SIZE = 100_000;
const WARMUP = 3;
const ITERATIONS = 20;

const s = new Series({ data: Array.from({ length: SIZE }, (_, i) => (i * 1.1) % 9999) });

for (let i = 0; i < WARMUP; i++) describe(s);

const times: number[] = [];
for (let i = 0; i < ITERATIONS; i++) {
  const t0 = performance.now();
  describe(s);
  times.push(performance.now() - t0);
}
const total = times.reduce((a, b) => a + b, 0);
console.log(JSON.stringify({ function: "series_describe", mean_ms: total / ITERATIONS, iterations: ITERATIONS, total_ms: total }));

File: benchmarks/pandas/bench_series_describe.py
"""Benchmark: Series.describe() — summary statistics on 100k Series."""
import json, time
import numpy as np
import pandas as pd

SIZE = 100_000
WARMUP = 3
ITERATIONS = 20

s = pd.Series((np.arange(SIZE) * 1.1) % 9999)
for _ in range(WARMUP): s.describe()

times = []
for _ in range(ITERATIONS):
    t0 = time.perf_counter()
    s.describe()
    times.append(time.perf_counter() - t0)
total = sum(times) * 1000
print(json.dumps({ "function": "series_describe", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total }))
